File: BeeCrowd/bee_2482.java
import java.util.Scanner;

class Idioma {
    String nome;
    String saudacao;

    public Idioma(String nome, String saudacao) {
        this.nome = nome;
        this.saudacao = saudacao;
    }
}

class Crianca {
    String nome;
    String idioma;

    public Crianca(String nome, String idioma) {
        this.nome = nome;
        this.idioma = idioma;
    }
}



public class bee_2482 {
    // Função para verificar se duas strings são iguais (conteúdo)
public static boolean stringsIguais(String a, String b) {
    if (a.length() != b.length()) return false;
    for (int i = 0; i < a.length(); i++) {
        if (a.charAt(i) != b.charAt(i)) return false;
    }
    return true;
}
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();  // número de traduções
        sc.nextLine();          // consumir quebra de linha

        Idioma[] idiomas = new Idioma[N];

        // Lê os idiomas e suas traduções
        for (int i = 0; i < N; i++) {
            String nomeIdioma = sc.nextLine();
            String saudacao = sc.nextLine();
            idiomas[i] = new Idioma(nomeIdioma, saudacao);
        }

        int M = sc.nextInt();  // número de crianças
        sc.nextLine();          // consumir quebra de linha

        Crianca[] criancas = new Crianca[M];

        // Lê o nome e idioma de cada criança
        for (int i = 0; i < M; i++) {
            String nome = sc.nextLine();
            String idioma = sc.nextLine();
            criancas[i] = new Crianca(nome, idioma);
        }

        // Para cada criança, encontra a saudação correspondente
        for (int i = 0; i < M; i++) {
            String saudacao = "";
            boolean encontrado = false;
            for (int j = 0; j < N && !encontrado; j++) { // só continua enquanto não encontrou
                if (stringsIguais(idiomas[j].nome, criancas[i].idioma)) {
                saudacao = idiomas[j].saudacao;
                encontrado = true; // marca como encontrado
        }
    }
            

            // Impressão da etiqueta
            System.out.println(criancas[i].nome);
            System.out.println(saudacao);
        }

        sc.close();
    }
}

File: BeeCrowd/bee_2482.c
#include <stdio.h>

typedef struct {
    char idioma[50];
    char saudacao[100];
} Traducao;

typedef struct {
    char nome[100];
    char idioma[50];
} Crianca;

int main() {
    int N, M;
    scanf("%d", &N);
    getchar(); // consumir o '\n' após o número

    Traducao traducoes[N];
    for (int i = 0; i < N; i++) {
        fgets(traducoes[i].idioma, 50, stdin);
        traducoes[i].idioma[strcspn(traducoes[i].idioma, "\n")] = 0; // remover '\n'
        fgets(traducoes[i].saudacao, 100, stdin);
        traducoes[i].saudacao[strcspn(traducoes[i].saudacao, "\n")] = 0; // remover '\n'
    }

    scanf("%d", &M);
    getchar(); // consumir o '\n'

    Crianca criancas[M];
    for (int i = 0; i < M; i++) {
        fgets(criancas[i].nome, 100, stdin);
        criancas[i].nome[strcspn(criancas[i].nome, "\n")] = 0; // remover '\n'
        fgets(criancas[i].idioma, 50, stdin);
        criancas[i].idioma[strcspn(criancas[i].idioma, "\n")] = 0; // remover '\n'
    }

    // Imprimir etiquetas
    for (int i = 0; i < M; i++) {
        char *saudacao = NULL;

        // Procurar a tradução correta
        for (int j = 0; j < N; j++) {
            if (strcmp(criancas[i].idioma, traducoes[j].idioma) == 0) {
                saudacao = traducoes[j].saudacao;
                break;
            }
        }

        // Imprimir etiqueta
        printf("%s\n", criancas[i].nome);
        printf("%s\n\n", saudacao); // linha em branco após a etiqueta
    }

    return 0;
}
